streamlink: 3.2.0 -> 4.1.0; pythonPackages.versioningit: init at 1.1.1

DeeUnderscore opened the pull request:
###### Description of changes
Changelogs:
* https://github.com/streamlink/streamlink/releases/tag/4.0.0
* https://github.com/streamlink/streamlink/releases/tag/4.0.1
* https://github.com/streamlink/streamlink/releases/tag/4.1.0

In 4.0.0, Streamlink switched to the PEP 518 format (pyproject.toml), and additionally introduced a build-time dependency on versioningit (although the PyPI source distribution has the version number substituted in, so we don't need the full Git history). 

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

DeeUnderscore committed:
pythonPackages.versioningit: init at 1.1.1

DeeUnderscore committed:
streamlink: 3.2.0 -> 4.1.0

Changelogs:
- https://github.com/streamlink/streamlink/releases/tag/4.0.0
- https://github.com/streamlink/streamlink/releases/tag/4.0.1
- https://github.com/streamlink/streamlink/releases/tag/4.1.0

Streamlink switched to the pyproject format in 4.0.0

fabaff commented on pkgs/development/python-modules/versioningit/default.nix:
```diff
@@ -0,0 +1,33 @@
+
+  format = "pyproject";
+
+  propagatedBuildInputs = [
+    importlib-metadata
+    packaging
+    setuptools
+    tomli
+  ];
```
```suggestion
  propagatedBuildInputs = [
    packaging
    setuptools
    tomli
  ] ++ lib.optionals (pythonOlder "3.10") [
    importlib-metadata
  ];
```

`dataclasses` is mentioned for Python < 3.7. I suggest to use `disabled = pythonOlder "3.7";` as everything else is EOL.

Could you please check if we can use `pytestCheckHook` to run the tests?

DeeUnderscore replied:
Okay, I was able to get the tests working properly (except some that want to write to Nix store)

fabaff commented on pkgs/development/python-modules/versioningit/default.nix:
```diff
@@ -0,0 +1,33 @@
+  ];
+
+  meta = with lib; {
+    description = "setuptools plugin for determining package version from VCS";
```
```suggestion
    description = "setuptools plugin for determining package version from VCS";
    homepage = "https://github.com/jwodder/versioningit";
```

DeeUnderscore commented:
[4.1.0](https://github.com/streamlink/streamlink/releases/tag/4.1.0) has been released, so I updated this PR accordingly. I've also rebased on master while at it.

DeeUnderscore commented:
@ofborg eval

wedens commented:
I can confirm that it works.

DeeUnderscore commented:
Closing in favor of #180910
